examples/mediadevices/src/helpers.js

```diff
@@ -14,42 +15,54 @@ function getDevicesOfKind(deviceInfos, kind) {
   });
 }
 
+function switchLocalTracks(room, newLockTrack) {
+  if (room) {
+      room.localParticipant.tracks.forEach(function(trackPublication) {
+      if (trackPublication.kind === newLockTrack.kind) {
+        room.localParticipant.unpublishTrack(trackPublication.track);
+      }
+    });
+    room.localParticipant.publishTrack(newLockTrack);
+  }
+}
+
 /**
  * Apply the selected audio input device.
  * @param {string} deviceId
  * @param {HTMLAudioElement} audio
+ * @param {Room} [room] to switch tracks on.
  * @returns {Promise<void>}
  */
-function applyAudioInputDeviceSelection(deviceId, audio) {
+function applyAudioInputDeviceSelection(deviceId, audio, room) {
   return Video.createLocalAudioTrack({
-    deviceId: deviceId
+    deviceId: {
+      exact: deviceId // NOTE: on ios safari - it respects the deviceId only if its exact.
+    }
   }).then(function(localTrack) {
     localTrack.attach(audio);
+    switchLocalTracks(room, localTrack);
+  }).catch(function(error) {
+    console.log('applyAudioInputDeviceSelection failed:', error);
   });
 }
 
-/**
- * Apply the selected audio output device.
- * @param {string} deviceId
- * @param {HTMLAudioElement} audio
- */
-function applyAudioOutputDeviceSelection(deviceId, audio) {
-  audio.setSinkId(deviceId);
-}
-
 /**
  * Apply the selected video input device.
  * @param {string} deviceId
  * @param {HTMLVideoElement} video
+ * @param {Room} [room] to switch tracks on.
```

Comment:
@makarandp0 `* @param {Room} [room] - The Room, if you have already joined one`?